zk_prover/src/merkle_sum_tree/mst.rs

```diff
@@ -25,9 +26,44 @@ pub struct MerkleSumTree<const N_ASSETS: usize, const N_BYTES: usize> {
     is_sorted: bool,
 }
 
-impl<const N_ASSETS: usize, const N_BYTES: usize> MerkleSumTree<N_ASSETS, N_BYTES> {
-    pub const MAX_DEPTH: usize = 29;
+impl<const N_ASSETS: usize, const N_BYTES: usize> Tree<N_ASSETS, N_BYTES>
+    for MerkleSumTree<N_ASSETS, N_BYTES>
+{
+    fn root(&self) -> &Node<N_ASSETS> {
+        &self.root
+    }
 
+    fn depth(&self) -> &usize {
+        &self.depth
+    }
+
+    fn leaves(&self) -> &[Node<N_ASSETS>] {
+        &self.nodes[0]
+    }
+
+    fn nodes(&self) -> &[Vec<Node<N_ASSETS>>] {
+        &self.nodes
+    }
+
+    /// Verifies a MerkleProof
+    fn verify_proof(&self, proof: &MerkleProof<N_ASSETS, N_BYTES>) -> bool
+    where
+        [usize; N_ASSETS + 1]: Sized,
+        [usize; 2 * (1 + N_ASSETS)]: Sized,
+    {
+        verify_proof(proof)
```

Comment:
I applied the changes in `b3990eb`. I believe that this can only apply to `verify_proof` while I cannot define the logic of `generate_proof` within the `Tree` trait as this will be different between `MerkleSumTree` and `AggregationMerkleSumTree`